### PR TITLE
fix oauth_encode()

### LINKED ISSUE #423: Token1.0 cannot generate proper oauth_signature params for UTF-8 characters
I was playing around [the demo of Twitter API](https://github.com/hadley/httr/blob/master/demo/oauth1-twitter.r) in httr package. Occasionally, I got an error when I tried to use Search API with UTF-8 characters.

```r
req <- GET("https://api.twitter.com/1.1/search/tweets.json",
           query = list(q = "写輪眼"),
           config(token = twitter_token))
#> Warning message:
#> In chars[!ok] <- unlist(lapply(chars[!ok], encode)) :
#>   number of items to replace is not a multiple of replacement length

status_code(req)
#> [1] 401

content(req)
#> $errors
#> $errors[[1]]
#> $errors[[1]]$code
#> [1] 32
#> 
#> $errors[[1]]$message
#> [1] "Could not authenticate you."
```

The cause of this seems `httr:::oauth_encode1()`, which mistakenly replaces original characters with less length of bytes. More precisely, here: https://github.com/hadley/httr/blob/1fc659856602f60ff75eb01903513244e3491ec2/R/oauth-signature.r#L105

```r
encoded_char <- httr:::oauth_encode1("写輪眼")
#> Warning message:
#> In chars[!ok] <- unlist(lapply(chars[!ok], encode)) :
#>   number of items to replace is not a multiple of replacement length

encoded_char
#> [1] "%8E%CA%97"

URLdecode(encoded_char)
#> [1] "写\x97"
```

The fix can be simple. I will send one later :)

### PR DESCRIPTION
**Pull request by yutannihilation**
This PR fixes #423.

The problem is in [this part](https://github.com/hadley/httr/blob/1fc659856602f60ff75eb01903513244e3491ec2/R/oauth-signature.r#L97) of `oauth_encode1()`:

```r
 encode <- function(x) paste0("%", toupper(as.character(charToRaw(x))))
```

This seems to expect the result of `charToRaw(x)` is a single character, but this assumption fails when `x` is a UTF-8 character:

```r
paste0(charToRaw("あ"))
[1] "82" "a0"
```

Therefore, [the following line](https://github.com/hadley/httr/blob/1fc659856602f60ff75eb01903513244e3491ec2/R/oauth-signature.r#L105) doesn't work well:

```r
 chars[!ok] <- unlist(lapply(chars[!ok], encode))
```

As the length of the left hand side is smaller than the right hand side, some bytes at the tail is lost when assigning. That's what this warning says.

```r
httr:::oauth_encode1("写輪眼")
#> [1] "写\x97"
#> Warning message:
#> In chars[!ok] <- unlist(lapply(chars[!ok], encode)) :
#>   number of items to replace is not a multiple of replacement length
```

This pull request adds `collapse = ""` so that `encode()` returns a single character vector.

### CODE REVIEW

**Comment by Crackz** [comment]
@yutannihilation  Thanks alot 🥇 you saved

**Comment by yutannihilation** [comment]
@Crackz Oh, I'm glad to hear that 😄 Actually, I've noticed this bug when I was trying rtweet...

**Comment by yutannihilation** [comment]
Ah, I got it. Hadley says this six months ago:

> I'm planning for httr2 in the next six months
https://github.com/hadley/httr/issues/408#issuecomment-260062341

**Comment by Crackz** [comment]
I hope it come soon  ... 
but he should pull requests at least 
i don't want to live with my customized one

**Comment by yutannihilation** [comment]
Agreed.

File: tests/testthat/test-oauth.R [hunk 1]
```diff
@@ -44,3 +44,14 @@ test_that("partial OAuth1 flow works", {
   expect_equal(status_code(r), 200)
 })
 
+test_that("oauth_encode1 works", {
+  # chinese characters for "xaringan"
+  orig_string <- "\u5199\u8f6e\u773c"
+  restored_string <- URLdecode(oauth_encode1(orig_string))
+  Encoding(restored_string) <- "UTF-8"
+
+  expect_equal(
```
Comment (hadley) [inline tests/testthat/test-oauth.R]: Can you please put this on one line?

**Comment by yutannihilation** [comment]
Thanks! Updated NEWS and test.

**Comment by hadley** [comment]
Thanks!

**Comment by yutannihilation** [comment]
Thanks for merging! This bug was a bit annoying, so I'm very happy to see this closed! 🍺

**Pull request merged**